webapp/publisher/snaps/views.py

```diff
@@ -202,6 +202,33 @@ def publisher_snap_metrics(snap_name):
     # to use 10, rather then latest
     default_track = helpers.get_default_track(snap_name)
 
+    annotations = {"name": "annotations", "series": [], "buckets": []}
+
+    for category in details["categories"]["items"]:
+        date = category["since"].split("T")[0]
+        new_date = logic.convert_date(category["since"])
+
+        if date not in annotations["buckets"]:
+            annotations["buckets"].append(date)
+
+        index_of_date = annotations["buckets"].index(date)
+
+        single_series = {
+            "values": [0] * (len(annotations)),
+            "name": category["name"],
+            "display_name": category["name"].capitalize().replace("-", " "),
```

Comment:
Not for this PR, but we need a common method to do that. It's already scattered across couple of places both in store and publisher parts.